spec/features/auto_deploy_spec.rb

```diff
@@ -413,11 +413,36 @@
 
   scenario 'receiving a `push` event for a deleted branch' do
     expect(github).to_not receive(:create_deployment)
-    push_event 'secret', deleted: true, head_commit: nil
+    push_event 'secret', deleted: true, head_commit: { id: "595ebd4ca061c4671ba89202aaf19c896f216635" }
```

Comment:
I think this breaks this test. When deleting a ref, github sends a push event where the head_commit id is nil. We need to be able to handle that scenario.